wpilibc/src/main/native/include/frc/apriltag/AprilTagFieldLayout.h

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <string_view>
+#include <vector>
+
+#include <wpi/SymbolExports.h>
+
+#include "frc/DriverStation.h"
+#include "frc/apriltag/AprilTag.h"
+#include "frc/geometry/Pose3d.h"
+
+namespace wpi {
+class json;
+}  // namespace wpi
+
+namespace frc {
+/**
+ * Class for representing a layout of AprilTags on a field and reading them from
+ * a JSON format.
+ *
+ * The JSON format contains a top-level "tags" field, which is a list of all
+ * AprilTags contained within a layout. Each AprilTag serializes to a JSON
+ * object containing an ID and a Pose3d.
+ *
+ * Pose3ds are assumed to be measured from the bottom-left corner of the field,
+ * when the blue alliance is at the left. Pose3ds will automatically be returned
+ * as passed in when calling GetTagPose(int).  Setting an alliance color via
+ * SetAlliance(DriverStation::Alliance) will mirror the poses returned from
+ * GetTagPose(int) to be correct relative to the other alliance.
+ */
+class WPILIB_DLLEXPORT AprilTagFieldLayout {
+ public:
+  AprilTagFieldLayout() = default;
+
+  /**
+   * Construct a new AprilTagFieldLayout with values imported from a JSON file.
+   *
+   * @param path Path of the JSON file to import from.
+   */
+  explicit AprilTagFieldLayout(std::string_view path);
+
+  /**
+   * Construct a new AprilTagFieldLayout from a vector of AprilTag objects.
+   *
+   * @param apriltags Vector of AprilTags.
+   */
+  explicit AprilTagFieldLayout(const std::vector<AprilTag>& apriltags);
```

Comment:
I checked the code out locally, and clang-tidy said to pass by value and move.
```suggestion
  explicit AprilTagFieldLayout(std::vector<AprilTag> apriltags);
```